src/main/java/pl/allegro/tech/discovery/consul/recipes/leader/LeaderElector.java

```diff
@@ -54,6 +55,8 @@
     private final int lockRescueDelaySeconds;
     private final LockAcquirer lockAcquirer;
 
+    private Disposable disposable;
```

Comment:
This field name doesn't describe what will be disposed when I call `dispose()`. I suggest sth like `watchCanceller` (see my comment in `Disposable` file).

Reply:
done